### PR TITLE
Update linters and rearrange `Gemfile` groups

### PR DESCRIPTION
**Pull request by bandogora**
I pretty much just copied over `.haml_lint.yml` and `.rubocop.yml` from round-three and added the `rubocop-performance` gem. I also moved some gems to more appropriate bundler groups in the `Gemfile`.

Edit: Almost forgot `eslint`.

### CODE REVIEW

File: .codeclimate.yml [hunk 1]
```diff
@@ -14,6 +14,7 @@ plugins:
     enabled: true
   rubocop:
     enabled: true
+    channel: rubocop-1-31-0
```
Comment (dirschn) [inline .codeclimate.yml:17]: image
From https://docs.codeclimate.com/docs/rubocop

Do you plan on adding one?

Comment (bandogora) [reply .codeclimate.yml:17]: We already have a `.rubocop.yml` that's compatible with that version

Comment (dirschn) [reply .codeclimate.yml:17]: Okay cool just checking

File: .rubocop.yml [hunk 2]
```diff
@@ -1,42 +1,84 @@
...
 
 AllCops:
   Exclude:
+    - './*'
     - 'bin/**/*'
-    - 'config/{environments,initializers}/**/*'
-    - 'config/{application,boot,environment,puma,spring}.rb'
+    - 'config/**/*'
     - 'db/**/*'
     - 'script/**/*'
```
Comment (dirschn) [inline .rubocop.yml:12]: There is no `script/` directory in this app

**Review by dirschn — APPROVED** [review]
this seems fine to me

**Pull request merged**